BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
 
-Create the following files somewhere on your local filesystem:
+This Dev Environment involves spinning up a virtual machine (most of the core contributors use VMWare, but Virtual Box should work as well) running Ubuntu 17.10 desktop.
 
-`config_api.toml`
-```toml
-[depot]
-path = "/hab/svc/builder-api/data"
-key_dir = "/path/to/bldr-key-pair"
+### Pre-Reqs
+
+* Ubuntu 17.10 Desktop Virtual Machine
+* Access to the "core" origin in [Production Builder](https://bldr.habitat.sh). Ask a core maintainer (or in the [Habitat Slack](http://slack.habitat.sh/)) if you do not have access.
+* Access to the Habitat 1Password account (NOTE - we are working on a solution to make this not required, but it is necessary for the time being)
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+```
+$ sudo visudo -f /etc/sudoers
+```
+
+### Builder Setup
+* Update your system
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Install curl
+```
+$ sudo apt-get install curl
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
+* Set up some environmental variables (I put these in my .bashrc)
+
+```
+export HAB_AUTH_TOKEN="<your github token>"
+export HAB_BLDR_URL="http://localhost:9636"
+export HAB_ORIGIN="core"
+```
+
+* symlink /src to path/to/your/habitat/repo
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Cd into your Habitat repo
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ cd path/to/your/habitat/repo
+```
+
+* Run the provision script
+
+```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
```

Comment:
I think `auth_token` should be already set after https://github.com/habitat-sh/habitat/blob/master/support/builder/dev-config.sh#L348 is run (called from https://github.com/habitat-sh/habitat/blob/master/support/linux/provision.sh#L35).

Quickly testing in a clean machine seems to confirm that `auth_token` is indeed set correctly as long as `export HAB_AUTH_TOKEN="<your github token>"` was run before.

I.e. can this part be removed?

Reply:
it does not work when I am testing, for some reason - even though I set HAB_AUTH_TOKEN beforehand.

Reply:
maybe with `sudo -E` ?

Reply:
I did set the sudo -E, and it still did not appear to work for me - but @baumanj has offered to run through these instructions again, if he has the same experience I will edit the instructions :)

Reply:
oh, wait, I'm sorry @mwrock - I misread your post above.  I will try that and see if using sudo -E make build-srv fixes the issue.

Reply:
There's a bit of a tricky situation here. If you export `HAB_BLDR_URL` _before_ running `provision.sh`, it will fail when it tries to run the [`hab pkg install` commands](https://github.com/habitat-sh/habitat/blob/master/support/linux/provision.sh#L23-L26). However, if you wait until after running `provision.sh` to add all the `HAB_` environment variables, you have to manually set the `auth_token` in `/hab/svc/builder-worker/config.toml`.

I see a couple options:
1. Set the `HAB_` env vars first and modify `provision.sh` to do its `hab pkg install`s with the `-u https://bldr.habitat.sh` option.
2. Set the `HAB_` env vars after `provision.sh` and set `auth_token` in `/hab/svc/builder-worker/config.toml` manually
3. Make `provision.sh` take the GitHub token as input and append the `HAB_` env var export lines to `~/.profile` _after_ the `hab pkg install`s.

Thoughts?

Reply:
Can we put the sudo inside the Makefile, so we don't need to type in a sudo -E ... every time to start up the env.